homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,132 @@
 from typing import Any
 
 from aiohttp import ClientConnectionError
-from intellifire4py import IntellifireAsync
+from intellifire4py import AsyncUDPFireplaceFinder, IntellifireAsync
 import voluptuous as vol
 
 from homeassistant import config_entries
 from homeassistant.const import CONF_HOST
-from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResult
 
-from .const import DOMAIN
+from .const import DOMAIN, LOGGER
 
 STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
 
 
-async def validate_input(hass: HomeAssistant, host: str) -> str:
+async def validate_host_input(host: str) -> str:
     """Validate the user input allows us to connect.
 
     Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
     """
     api = IntellifireAsync(host)
     await api.poll()
-
+    ret = api.data.serial
+    LOGGER.debug("Found a fireplace: %s", ret)
     # Return the serial number which will be used to calculate a unique ID for the device/sensors
-    return api.data.serial
+    return ret
 
 
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for IntelliFire."""
 
     VERSION = 1
 
-    async def async_step_user(
+    def __init__(self):
+        """Initialize the Config Flow Handler."""
+        self._config_context = {}
+        self._discovered_hosts: list[str] = []
+
+    async def _find_fireplaces(self):
+        """Perform UDP discovery."""
+        fireplace_finder = AsyncUDPFireplaceFinder()
+        self._discovered_hosts = await fireplace_finder.search_fireplace(timeout=1)
+
+    async def async_step_local_config(self, user_input=None):
+        """Handle local ip configuration."""
+
+        errors = {}
+        placeholder = {}
```

Comment:
```suggestion
```

This isn't used. There are no placeholders in the description

Reply:
<img width="748" alt="image" src="https://user-images.githubusercontent.com/6491743/158004541-92ec0482-58ab-4401-aae0-9a4f484e0a70.png">
<img width="515" alt="image" src="https://user-images.githubusercontent.com/6491743/158004544-afd83b24-9647-4ff0-b5d3-f058eb29833a.png">

Reply:
Yes there are -> its used for this error state.